apps/sms/test/marionette/share_activity_test.js

```diff
@@ -33,105 +34,108 @@ marionette('Messages as share target', function() {
+          conversationView.headerAction,
+          'close',
           'Close activity button should be visible'
         );
 
-        // Exit from activity and verify that Messages is dismissed
-        messagesApp.performHeaderAction();
-        messagesApp.waitForAppToDisappear();
+        newMessageView.back();
       });
 
       test('Should return to Conversation view if in Report view', function() {
         // Send message to be forwarded to Conversation panel afterwards
-        messagesApp.addRecipient('+1');
-        messagesApp.send();
+        newMessageView.addNewRecipient('+1');
+        var conversationView = newMessageView.send();
 
-        // Go to the Report panel
-        messagesApp.contextMenu(messagesApp.Conversation.message);
-        messagesApp.Menu.selectAppMenuOption('View message report');
-        messagesApp.Report.waitToAppear();
+        var reportView = conversationView.openReport(
+          conversationView.messages()[0].id
+        );
 
         assert.equal(
-          messagesApp.Report.header.getAttribute('action'), 'close',
+          reportView.headerAction,
+          'close',
           'Close activity button should be visible'
         );
 
         // Close report panel
-        messagesApp.performReportHeaderAction();
-        client.helper.waitForElement(messagesApp.Conversation.message);
+        reportView.back();
 
         assert.equal(
-          messagesApp.NewMessage.header.getAttribute('action'), 'close',
+          conversationView.headerAction,
+          'close',
           'Close activity button should be visible'
         );
 
-        // Exit from activity and verify that Messages is dismissed
-        messagesApp.performHeaderAction();
-        messagesApp.waitForAppToDisappear();
+        conversationView.back();
       });
 
-      // Temporarily skip due to intermittent failures - Participants view is
-      // not opened after tapping on the conversation header.
-      test.skip('Should return to Conversation panel if in Participants panel',
+      test('Should return to Conversation panel if in Participants panel',
       function() {
         // Send group MMS message
-        messagesApp.addRecipient('+1');
-        messagesApp.addRecipient('+2');
+        newMessageView.addNewRecipient('+1');
+        newMessageView.addNewRecipient('+2');
 
         // TODO: This should be moved to the appropriate test case in:
         // https://bugzilla.mozilla.org/show_bug.cgi?id=1043903
-        messagesApp.Composer.messageInput.sendKeys(
+        newMessageView.typeMessage(
           'very_very_very_very_very_very_long_message_i_can_ever_imagine'
         );
 
-        messagesApp.send();
-
         // Go to Participants panel
-        messagesApp.Conversation.headerTitle.tap();
-        messagesApp.Participants.waitToAppear();
+        var conversationView = newMessageView.send();
+
+        assert.equal(conversationView.headerTitle, '+1\n (+1)');
```

Comment:
Grr, I'll remove this assert if we land this PR - just forgot to remove :/ (Though I was quite surprised that we have ```\n``` here :))